src/unix/stream.c

```diff
@@ -1180,6 +1180,9 @@ static void uv__read(uv_stream_t* stream) {
       } else if (errno == ECONNRESET && stream->type == UV_NAMED_PIPE) {
         uv__stream_eof(stream, &buf);
         return;
+#elif defined(_AIX)
```

Comment:
Is this actually AIX specific or is it only that we've seen the problem on AIX?

Reply:
I have tried reproducing it with ubuntu on x64 but `read()` returns `0` even when the connection is reset, seems like AIX is less forgiving and returns `-1`. Having said that, this PR https://github.com/libuv/libuv/commit/05a003a3f78d07185b7137601fe8e93561855a8d had a similar check added to every architecture but then got reverted. We can also add this globally or only on AIX.

Reply:
Also a similar code exists above this line specific to windows platforms.

Reply:
The revert (https://github.com/libuv/libuv/commit/01544d861b9e11f844d482a80bc557dd9b803c8c) was because a Node.js test failed (https://github.com/nodejs/node/pull/2310#issuecomment-128279071) so we should have a libuv-in-node CI run to check if the regression reoccurs.

Reply:
We have a libuv+node.js integration test nowadays so we can test whether it causes regressions.

Reply:
@bnoordhuis, @cjihrig  do you think we should leave as adding only for AIX or try to make it more generic. I'm good either way.

Reply:
If the behavior hasn't been observed on other platforms then I'm good with the #ifdef.